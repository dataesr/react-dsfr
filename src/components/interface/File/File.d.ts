import * as React from 'react';

export interface FileProps {
  className?: string;
  label: string;
  multiple?: boolean;
  onChange?: (...args: any[])=>any;
  errorMessage?: string;
  hint?: string;
}

declare const File: React.FC<FileProps>;

export default File;
