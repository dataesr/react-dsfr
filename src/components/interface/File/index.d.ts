import type File from './File';

export type { File };
