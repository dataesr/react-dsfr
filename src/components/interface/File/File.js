import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import dataAttributes from '../../../utils/data-attributes';

/*
* DSFR v1.2
*/
import '@gouvfr/dsfr/dist/component/link/link.css';

/**
 *
 * @visibleName Link
 */
const File = ({
  className,
  label,
  errorMessage,
  hint,
  onChange,
  multiple,
  ...remainingProps
}) => {
  const _className = classNames(
    'fr-upload-group',
    className, {
      [`ds-fr--${label}`]: label,
    },
  );

  return (
    <div {...dataAttributes.getAll(remainingProps)} className={_className}>
      <label className="fr-label" htmlFor="file-upload">
        {label}
        {hint && <span className="fr-hint-text">{hint}</span>}
      </label>
      <input
        onChange={onChange}
        className="fr-upload"
        type="file"
        aria-describedby={hint || undefined}
        multiple={multiple}
      />
      {errorMessage && (
      <p id="file-upload-with-error-desc-error" className="fr-error-text">
        {errorMessage}
      </p>
      )}
    </div>
  );
};

File.defaultProps = {
  className: '',
  hint: '',
  errorMessage: '',
  multiple: false,
  onChange: () => {},
};

File.propTypes = {
  className: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
    PropTypes.array,
  ]),
  label: PropTypes.string.isRequired,
  multiple: PropTypes.bool,
  onChange: PropTypes.func,
  errorMessage: PropTypes.string,
  hint: PropTypes.string,
};

export default File;
