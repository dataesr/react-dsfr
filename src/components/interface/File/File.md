## Ajout de fichier - File upload

```jsx
<File
    onChange={() => {}}
    multiple
    label="Label File"
    hint="Hint"
    errorMessage="Format de fichier non supportÃ©"
/>
```
