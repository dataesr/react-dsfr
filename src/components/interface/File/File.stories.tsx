import { ComponentStory } from '@storybook/react';

import File from './File';

export default {
  title: 'File',
  component: File,
  subcomponents: { File },
};

const Template: ComponentStory<typeof File> = (args) => (
  <File {...args} />
);

export const SimpleFile = Template.bind({});
SimpleFile.args = {
  label: 'Label File',
  hint: 'Hint',
};
SimpleFile.storyName = 'Ajout de fichier';

export const AcceptFile = Template.bind({});
AcceptFile.args = {
  label: 'Label File',
  hint: 'Hint',
  accept: '.txt, .csv',
};
AcceptFile.storyName = 'Ajout de fichier specifique';

export const MultipleFile = Template.bind({});
MultipleFile.args = {
  multiple: true,
  label: 'Label File',
  hint: 'Hint',
};
MultipleFile.storyName = 'Ajout de fichier multiples';

export const ErrorFile = Template.bind({});
ErrorFile.args = {
  label: 'Label File',
  hint: 'Hint',
  errorMessage: "Message d'erreur",
};
ErrorFile.storyName = 'Ajout de fichier en erreur';
