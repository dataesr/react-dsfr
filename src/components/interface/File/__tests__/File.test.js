import Enzyme, { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from '@cfaester/enzyme-adapter-react-18';
import File from '../File';

Enzyme.configure({ adapter: new Adapter() });

describe('<File />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = (props = {}) => shallow(
      <File label={props.label} onChange={props.onChange} />,
    );
  });

  it('renders correctly', () => {
    const component = renderer
      .create(
        <File label="Label File" hint="Hint" message="Format de fichier non supportÃ©" />,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  test('OnChange File', () => {
    const onChange = jest.fn();
    const props = {
      label: 'label',
      onChange,
    };

    const component = wrapper({ ...props });
    component.find('input').simulate('change');
    expect(onChange).toHaveBeenCalled();
  });
});
