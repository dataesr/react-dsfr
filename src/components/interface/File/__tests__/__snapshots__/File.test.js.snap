// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<File /> renders correctly 1`] = `
<div
  className="fr-upload-group ds-fr--Label File"
>
  <label
    className="fr-label"
    htmlFor="file-upload"
  >
    Label File
    <span
      className="fr-hint-text"
    >
      Hint
    </span>
  </label>
  <input
    aria-describedby="Hint"
    className="fr-upload"
    multiple={false}
    onChange={[Function]}
    type="file"
  />
  
</div>
`;
